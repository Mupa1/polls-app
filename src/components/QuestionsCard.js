import React from 'react';

const QuestionsCard = () => (
  <article>
    <div>
      <h2>Title</h2>
      <p>Choice</p>
      <p>published_at</p>
    </div>
  </article>
);

export default QuestionsCard;
