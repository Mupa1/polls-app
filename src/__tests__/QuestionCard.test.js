/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import store from '../redux/store';
import QuestionsCard from '../components/QuestionsCard';

const props = {
  title: 'Favourite programming language?',
  published_at: '2021-07-29T22:25:29.739Z',
  choices: 7,
  url: '/questions/34',
};

describe('QuestionsCard component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <Provider store={store}>
          <QuestionsCard {...props} />
        </Provider>
      </BrowserRouter>, div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <QuestionsCard {...props} />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
