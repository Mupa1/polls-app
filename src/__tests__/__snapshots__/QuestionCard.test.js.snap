// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsCard component matches snapshot 1`] = `
<a
  href="/questions/34"
  onClick={[Function]}
>
  <article
    className="question-card"
  >
    <h2
      className="title"
    >
      Favourite programming language?
    </h2>
    <p
      className="published-on"
    >
      Published on:
       
      Fri Jul 30 2021
    </p>
    <p>
      7
       
      Choices
    </p>
  </article>
</a>
`;
